sqe-http-api/Controllers/ArtefactController.cs

```diff
@@ -24,20 +22,48 @@ public ArtefactController(IUserService userService, IArtefactService artefactSer
             this._userService = userService;
         }
         
+        private void ParseOptionals(List<string> optionals, out bool images, out bool masks)
+        {
+            images = masks = false;
+            if (optionals == null) return;
+            images = optionals.Contains("artefacts");
+            masks = optionals.Contains("masks");
+        }
+        
+        /// <summary>
+        /// Provides a listing of all artefacts that are part of the specified edition
+        /// </summary>
+        /// <param Name="editionId">Unique Id of the desired edition</param>
+        /// <param Name="artefactId">Unique Id of the desired artefact</param>
+        /// <param Name="images">Defines whether image references should be included in the response</param>
+        [AllowAnonymous]
+        [HttpGet("edition/{editionId}/artefact/{artefactId}")]
+        public async Task<ActionResult<ArtefactDTO>> GetArtefact([FromRoute] uint editionId, [FromRoute] uint artefactId, [FromQuery] List<string> optional = null)
+        {
+            ParseOptionals(optional, out var images, out var masks);
+            try
+            {
+                return await _artefactService.GetEditionArtefactAsync(_userService.GetCurrentUserObject(), artefactId, masks);
+            }
+            catch (NotFoundException)
+            {
+                return NotFound();
+            }
+        }
+        
         /// <summary>
         /// Provides a listing of all artefacts that are part of the specified edition
         /// </summary>
         /// <param Name="editionId">Unique Id of the desired edition</param>
         /// <param Name="images">Defines whether image references should be included in the response</param>
         [AllowAnonymous]
         [HttpGet("edition/{editionId}/artefact/list")]
-        public async Task<ActionResult<ArtefactListDTO>> GetArtefacts([FromRoute] uint editionId, [FromQuery] string images = "false")
+        public async Task<ActionResult<ArtefactListDTO>> GetArtefacts([FromRoute] uint editionId, [FromQuery] List<string> optional = null)
```

Comment:
Actually, there's no point in specifying a null default value. ASP.NET Core's default binder passes an empty list by default. At least that's what happened when I tried it.

Reply:
Thanks, you must be right.  I have changed all 3 occurences.